### PR TITLE
OY-4152 Replace draft.js with Lexical

### PR DESCRIPTION
**Pull request by vaeinoe**
As draft.js support has ended, replace it with a custom editor based on Meta Lexical editor framework and heavily customized & stripped down Lexical sandbox code.

**Outstanding bugs that have to be fixed before release:**

- [x] Language versions do not work correctly: editing single language version updates multiple languages
- [x] Line breaks / paragraphs sometimes get doubled when saving
- [x] Activation hot spot / active text editor zone is too small on an empty editor, causing people to think the editor can't be selected

**Future improvements:**

- MUI icons are used and text styles matched, but there's still improvement for removing style duplication or using the generic components if there's need. Editor theming doesn't support styled-components properly, so there still has to be _some_ duplication.

**Things to test / consider:**

- ~**HTML serialization & deserialization may still not always match draft.js input and output 1:1**, unit tests pass and everything seems OK but this needs to be thoroughly tested manually~ one issue with line breaks fixed, no further ones reported from the field so far
- ~There was a request to add **link alt text functionality** - I'm not sure why text links need that, needs to be discussed~ seems like a misunderstanding
- ~Does **HTML copy & paste** to and from clipboard need to be supported and if so, to which degree?~ no requests during testing period
- ~Does the editor need to **support any additional HTML styles** that aren't available in the old or new editor UIs?~ no issues / requests during testing period

### CODE REVIEW

File: src/main/app/src/components/LexicalEditorUI/LexicalEditorUI.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+          )}
+        </>
+        <HistoryPlugin />
+        <OnChangePlugin onChange={onChange} />
+      </LexicalComposer>
+    </Container>
+  );
+};
+
+export default LexicalEditorUI;
```
Comment (SalamaGofore) [inline src/main/app/src/components/LexicalEditorUI/LexicalEditorUI.tsx:97]: Ehkä tämän default exportin voisi jättää pois

File: src/main/app/src/components/LexicalEditorUI/ui/DropDown.tsx [hunk 2]
```diff
@@ -0,0 +1,233 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
```
Comment (SalamaGofore) [inline src/main/app/src/components/LexicalEditorUI/ui/DropDown.tsx:8]: Onko tässä muutoksia toiminnallisuuteen?

**Review by SalamaGofore — CHANGES_REQUESTED** [review]

File: src/main/app/src/components/LexicalEditorUI/plugins/ToolbarPlugin/Components.tsx [hunk 3]
```diff
@@ -0,0 +1,177 @@
+import styled from 'styled-components';
+
```
[the remark this answers is not in the document]
Comment (vaeinoe) [reply src/main/app/src/components/LexicalEditorUI/plugins/ToolbarPlugin/Components.tsx:2]: Tässä on tyylitelty sen verran pluginia uusiksi ja wrapattu koko komeus styled-componentsiin, että lisenssiä ei taida kannata lisätä. Tuolta toisaalta kyllä oli tippunut pois.

File: src/main/app/src/components/LexicalEditorUI/plugins/FloatingLinkEditorPlugin/index.tsx [hunk 4]
```diff
@@ -0,0 +1,323 @@
+/* eslint-disable no-negated-condition */
```
[the remark this answers is not in the document]
Comment (vaeinoe) [reply src/main/app/src/components/LexicalEditorUI/plugins/FloatingLinkEditorPlugin/index.tsx:1]: Merkitsin olennaiset muutokset headeriin.

File: src/main/app/src/components/LexicalEditorUI/plugins/ToolbarPlugin/index.tsx [hunk 5]
```diff
@@ -0,0 +1,347 @@
+/* eslint-disable no-negated-condition */
+
```
[the remark this answers is not in the document]
Comment (vaeinoe) [reply src/main/app/src/components/LexicalEditorUI/plugins/ToolbarPlugin/index.tsx:16]: Tarvitsi, oli tippunut pois. Merkitsin myös kommentilla olennaiset muutokset.

File: src/main/app/src/components/LexicalEditorUI/ui/DropDown.tsx (hunk 2 above)
Comment (vaeinoe) [reply src/main/app/src/components/LexicalEditorUI/ui/DropDown.tsx:8]: Tässä oli vain wrapattu tyylit styled-componentsilla ja siistitty + viilattu tyylejä hiukan. Merkitsin tähänkin kommentilla.

**Review by SalamaGofore — APPROVED** [review]

**Pull request merged**